packages/projects-docs/pages/learn/getting-started/setting-up-repository.mdx

```diff
@@ -8,43 +8,60 @@ import { Callout } from 'nextra-theme-docs'
 
-The first time when you import a repo, it can take a while. During first-time setup, we clone the repository into the VM and set up the folders for the project. However, after this setup we will create a memory snapshot of the VM, which will ensure that next VM loads (& clones!) will be fast. Now, let's take a look at editor setup!
+The first time when you import a repo, it can take a while. During first-time setup, we clone the repository into the VM and set up the folders for the project. However, after this setup, we will create a memory snapshot of the VM, which will ensure that all the following VM loads (& clones!) will be fast.
+
+Next, let's take a look at the editor setup!
 
 ## Editor & Workflow Setup
 
-CodeSandbox has an opinionated flow when it comes to working with repositories, and you can see hints of this in the editor. As soon as you enter the editor, you will see the current branch name in the header. Every branch has its own VM and is available through a unique URL. For example, if two people visit the URL for the `main` branch, they will be able to see the cursor, terminals and devtools of each-other.
+CodeSandbox has an opinionated flow when it comes to working with repositories, and you can see hints of this in the editor.
+
+As soon as you enter the editor, you will see the current branch name in the header. Every branch has its own VM and is available through a unique URL. For example, if two people visit the URL for the `main` branch, they will be able to see each other's cursor, terminals and DevTools.
 
-<Callout emoji="💡">URLs for branches are predictable, if you want to open a specific branch, you can update the URL with thew new branch name, and we'll import & create an environment for that branch automatically.</Callout>
+<Callout emoji="💡">URLs for branches are predictable. So, if you want to open a specific branch, you can update the URL with the new branch name, and we'll import & create an environment for that branch automatically.</Callout>
 
 Now it's time to write some code!
 
-The default branch is protected by default. This means that the branch is read-only, and will sync automatically with the branch on GitHub when new commits come in. To actually write some code, you need to create a new branch. You can do this by either typing some code, or by clicking the “Branch” button in the header. This will automatically create a new branch and a VM tied to that branch.
+### Branching Workflow
+
+On CodeSandbox, the `main` branch is protected by default. This means that it's read-only and will sync automatically with the branch on GitHub when new commits come in.
+
+To write some code, you have three options:
+1. (Recommended) Just start typing some code - we will create a new branch for you using seamless branching!
+2. Click the “Branch” button in the header.
+3. [Toggle protection off](/learn/environment/branch-protection) for the `main` branch.
+
+Give it a try!
```

Comment:
Should this be a link to something - like a OS repo?
I feel like if we are encouraging people to try something, it should be accompanied by an action.

Reply:
My train of thought here is that users reading through the guide have likely imported a repo at the beginning, following our prompt:

> Once you've created the team, you can import a repository by clicking “Import from GitHub” and entering the URL of the GitHub repo. Once you've clicked “Import”, you're off to the races!

So, at this stage, users should likely be on the main branch of their newly-imported repo. However, we can make that more evident by changing the last sentence to:

> Give it a try on the repository you imported!